collector/pg_stat_progress_vacuum.go

```diff
@@ -0,0 +1,202 @@
+		prometheus.BuildFQName(namespace, progressVacuumSubsystem, "index_vacuum_count"),
+		"Number of completed index vacuum cycles.",
+		[]string{"datname", "relname"},
+		nil,
+	)
+
+	statProgressVacuumMaxDeadTuples = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, progressVacuumSubsystem, "max_dead_tuples"),
+		"Maximum number of dead tuples that can be stored before cleanup is performed.",
+		[]string{"datname", "relname"},
+		nil,
+	)
+
+	statProgressVacuumNumDeadTuples = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, progressVacuumSubsystem, "num_dead_tuples"),
+		"Current number of dead tuples found so far.",
+		[]string{"datname", "relname"},
+		nil,
+	)
+
+	// This is the view definition of pg_stat_progress_vacuum, albeit without the conversion
+	// of "phase" to a human-readable string. We will prefer the numeric representation.
+	statProgressVacuumQuery = `SELECT
+		d.datname,
+		s.relid::regclass::text AS relname,
+		s.param1 AS phase,
+		s.param2 AS heap_blks_total,
+		s.param3 AS heap_blks_scanned,
+		s.param4 AS heap_blks_vacuumed,
+		s.param5 AS index_vacuum_count,
+		s.param6 AS max_dead_tuples,
+		s.param7 AS num_dead_tuples
+	FROM
+		pg_stat_get_progress_info('VACUUM'::text)
+		s(pid, datid, relid, param1, param2, param3, param4, param5, param6, param7, param8, param9, param10, param11, param12, param13, param14, param15, param16, param17, param18, param19, param20)
+	LEFT JOIN
+		pg_database d ON s.datid = d.oid`
+)
+
+func (c *PGStatProgressVacuumCollector) Update(ctx context.Context, instance *instance, ch chan<- prometheus.Metric) error {
+	db := instance.getDB()
+	rows, err := db.QueryContext(ctx,
+		statProgressVacuumQuery)
+
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var (
+			datname          sql.NullString
+			relname          sql.NullString
+			phase            sql.NullInt64
+			heapBlksTotal    sql.NullInt64
+			heapBlksScanned  sql.NullInt64
+			heapBlksVacuumed sql.NullInt64
+			indexVacuumCount sql.NullInt64
+			maxDeadTuples    sql.NullInt64
+			numDeadTuples    sql.NullInt64
+		)
+
+		if err := rows.Scan(
+			&datname,
+			&relname,
+			&phase,
+			&heapBlksTotal,
+			&heapBlksScanned,
+			&heapBlksVacuumed,
+			&indexVacuumCount,
+			&maxDeadTuples,
+			&numDeadTuples,
+		); err != nil {
+			return err
+		}
+
+		datnameLabel := "unknown"
+		if datname.Valid {
+			datnameLabel = datname.String
+		}
+		relnameLabel := "unknown"
+		if relname.Valid {
+			relnameLabel = relname.String
+		}
+
+		labels := []string{datnameLabel, relnameLabel}
+
+		phaseMetric := 0.0
+		if phase.Valid {
+			phaseMetric = float64(phase.Int64)
+		}
+		ch <- prometheus.MustNewConstMetric(statProgressVacuumPhase, prometheus.GaugeValue, phaseMetric, labels...)
+
+		heapTotal := 0.0
+		if heapBlksTotal.Valid {
+			heapTotal = float64(heapBlksTotal.Int64)
+		}
+		ch <- prometheus.MustNewConstMetric(statProgressVacuumHeapBlksTotal, prometheus.GaugeValue, heapTotal, labels...)
```

Comment:
If this is a true monotonic counter, it should be exposed as such.

```suggestion
		ch <- prometheus.MustNewConstMetric(statProgressVacuumHeapBlksTotal, prometheus.CounterValue, heapTotal, labels...)
```